tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/IConnection.cs

```diff
@@ -0,0 +1,23 @@
+// <copyright file="IConnection.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.ComponentModel;
+using System.Net;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb
+{
+    /// <summary>
+    /// MongoDB.Driver.Core.IConnection interface for duck-typing
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public interface IConnection
```

Comment:
I think this could be a `[DuckCopy]` struct, though then we can't use it as a constraint in the integrations, so would need to do manual duck typing in `MongoDbIntegration.CreateScope()`. 

Pros of struct: Would only do the duck type if the integration is enabled/not already in mongo context, less overhead
Cons: More faff, not sure if the type constraint approach would have any performance benefits @tonyredondo, or if it's just convenience?

Reply:
The constraints approach have similar performance than the DuckCopy but allowing methods calls. This because internally the generic T is an struct type. I prefer to use the constraint approach if we can, the limitation is that we cannot compare with null unless we also apply the IDuckType interface as a constraint.

Reply:
@tonyredondo can you clarify, what is your recommendation here? Should I keep as is or make this a struct?

Reply:
@tonyredondo I don't quite follow either. The two options AFAIK are:

1. Use interfaces as above, and use constraint in the integration. If I understand you correctly, you're saying that the actual type created will be a struct that implements the proxy interface, so we get the "best of both worlds" in that case - it's a struct, but we can also call methods (which we can't do with a `[DuckCopy]` struct
2. As this type doesn't use any methods, we can a `[DuckCopy]` struct, but we can't use it in a constraint, so would need to do the duck typing _inside_ the integration, instead of as part of the CallTarget functionality. 

Presumably, in this case, the performance is similar (I assume we don't `[DuckCopy]` if it's a "constraint struct", does that matter?), and in some cases (where we need to call methods) 1. will be the only option. In which case I can't really see a case where 2. would be preferable?

>the limitation is that we cannot compare with null unless we also apply the IDuckType interface as a constraint

Is there a downside to applying the `IDuckType` constraint, as the type always implements it, correct?

Reply:
I created this PR, with the best practices and benchmarks for all options:

https://github.com/DataDog/dd-trace-dotnet/pull/2223

Reply:
Looks like the best way to handle this is to keep the interface with generic constraint, so I'll keep this type as-is